### PR TITLE
Refactored and added step for publishing docs and website to circleci

### LINKED ISSUE #954: add step to circleci to generate scaladoc and include in website
Circleci changes:

- generate scaladoc
- make sure scaladoc published as part of website step

### PR DESCRIPTION
**Pull request by softinio**
# Resolution

This resolves #954 

# Post this PR and Post next release TODO

#955 

# References relevant to this PR

- https://scalameta.org/mdoc/docs/docusaurus.html
- https://docusaurus.io/docs/en/publishing#using-circleci-20

### CODE REVIEW

**Comment by softinio** [comment]
@ghostdogpr Saw your comment. I am hoping this PR will fix netlify too.

My plan is to deploy to github pages and leave netlify pointing to github pages for now. We can improve on this flow going forward.

**Comment by ghostdogpr** [comment]
@softinio yep, the workflow should be the same. It’s just that I uploaded manually the content of the `zio` folder at the root of the branch directly, and I’m not sure if the build will do the same or upload the folder itself, in which case we have to fix what we point at from Netlify. Should be an easy fix.

File: .circleci/config.yml [hunk 1]
```diff
@@ -101,26 +101,49 @@ jobs:
...
+    docker:
+      - image: hseeberger/scala-sbt:8u181_2.12.8_1.2.8
+    steps:
+      - checkout
+      - restore_cache:
+          key: sbt-cache
+      - run:
+          name: Generating docs
+          command: |
+            sbt coreJVM/doc coreJS/doc interopCatsJVM/doc interopCatsJS/doc interopFutureJVM/doc interopScalaz7xJVM/doc interopScalaz7xJS/doc interopMonixJVM/doc interopMonixJS/doc interopJavaJVM/doc interopReactiveStreamsJVM/doc interopTwitterJVM/doc
```
Comment (regiskuckaertz) [inline .circleci/config.yml]: Can you please include `streamsJVM` as well?

**Comment by softinio** [comment]
@regiskuckaertz made change you requested (good idea!) can I trouble you to re-review when you have a moment? Thanks!

**Comment by regiskuckaertz** [comment]
🤠

**Pull request merged**